freqtrade/freqtradebot.py

```diff
@@ -226,15 +218,16 @@ def get_target_bid(self, pair: str, ticker: Dict[str, float]) -> float:
             logger.debug('order_book %s', order_book)
             # top 1 = index 0
             order_book_rate = order_book['bids'][order_book_top - 1][0]
-            # if ticker has lower rate, then use ticker ( usefull if down trending )
             logger.info('...top %s order book buy rate %0.8f', order_book_top, order_book_rate)
-            if ticker_rate < order_book_rate:
```

Comment:
While looking at the tests for this (`test_order_book_bid_strategy3` in particular) I noticed that this is a potentially heavy logic change.

The old logic uses the ticker if that has a lower rate. I'm just making sure that this is an intendet change.

This is not per se a problem, since i'm not convinced that `get_ticker` is always fully uptodate on exchanges (most discourage using this function to get the latest price, but encourage using orderbook l2 instead) as pointed out [here](https://ccxt.readthedocs.io/en/latest/manual.html?highlight=fetchTicker#notes-on-ticker-structure)
> Although some exchanges do mix-in orderbook’s top bid/ask prices into their tickers (and some even top bid/asks volumes) you should not treat a ticker as a fetchOrderBook replacement. The main purpose of a ticker is to serve statistical data, as such, treat it as “live 24h OHLCV”. It is known that exchanges discourage frequent fetchTicker requests by imposing stricter rate limits on these queries. If you need a unified way to access bid/asks you should use fetchL[123]OrderBook family instead.

Reply:
> The old logic uses the ticker if that has a lower rate. I'm just making sure that this is an intendet change.

The problem here is that if we, as stated in the ccxt manual, are going to use 'a unified way to access bid/asks', i.e. order book and set use_order_book: true in the config, the application will fetch the ticker anyway (in order to compare it with the order_book_rate), *BUT* on some exchanges (BitMex) ticker does not contain bid and ask prices, so the current logic fails. The only way to make it work with order book data only is to remove this strange comparison, which was proposed in this PR.

As a possible alternative, an additional setting (errm, 'compare_ticker_rate_with_order_book_rate: true (default)/false') could be added into config. This will make all tests happy *and* will allow to use order book rate without requesting ticker.

Reply:
it's NOT about making tests happy (it never was).

You gave it some thought and intentionally decided to remove it - which I'm fine with.

not 100% sure the comparison made sense anyway, as I think the only way this can happen is if the ticker is a) outdated or b) plain outright wrong, since orderbook should be the latest version.

Reply:
Well, I also don't know of usefulness of this comparison and do not insist on removing it.
I only see the claim from the ccxt that ticker is a kind of statistical data and would like to allow 2 modes for FT: either use ticker data only (for instance, if exchange do not provide access to order book) *or* use order book data only (for instance, if ticker is buggy or has strong antispam limits for particular exchenge).

If you would not like to change current logic that includes this comparison, I proposed above a workaround with additional setting in the config which will control it.

Reply:
you had me convinced already :)

ccxt documentation agrees with removing it - and so do i.